tests/lib/rendererCoverageCollector.js

```diff
@@ -0,0 +1,180 @@
+    if (window.__coverage__) {
+        var xhr = new XMLHttpRequest();
+        xhr.onreadystatechange = function () {
+            if (this.readyState === 4) {
+                if (this.status === 200) {
+                    console.log("Saved coverage data.");
+                }
+                else {
+                    console.error("Error saving coverage data:", this.responseText);
+                }
+
+                // notify the main process TODO or listen for even of server
+                notifyCoverageSuccess();
+            }
+        };
+        xhr.open("POST", "http://localhost:" + coveragePort + "/coverage");
+        xhr.setRequestHeader("Content-type", "application/json");
+        var wrappedPayload = {
+            payload: {
+                document: {
+                    title: document.title,
+                    URL: document.URL
+                },
+                navigator: {
+                    appCodeName: navigator.appCodeName,
+                    appName: navigator.appName,
+                    product: navigator.product,
+                    productSub: navigator.productSub,
+                    userAgent: navigator.userAgent,
+                    vendor: navigator.vendor,
+                    vendorSub: navigator.vendorSub
+                },
+                coverage: window.__coverage__
+            }
+        };
+        xhr.send(JSON.stringify(wrappedPayload, null, 2));
+    } else {
+        // notify anyways
+        notifyCoverageSuccess();
+    }
+};
+
+
+
+/**
+ * Component that ensures `gpii.app.dialog`s are using the instrumented version of renderer content
+ */
+fluid.defaults("gpii.tests.app.instrumentedDialog", {
+    // use the instrumented renderer files
+    config: {
+        filePrefixPath: "/instrumented/src/renderer"
+    },
+    listeners: {
+        // "onDestroy.cleanupElectron": null,
+        // "onCreate.attachCoverageCollector": {
+        //     funcName: "gpii.tests.app.instrumentedDialog.attachCoverageCollector",
+        //     args: ["{that}.dialog"]
+        // },
+        "onCreate.log": { // XXX dev
+            funcName: "console.log",
+            args: ["INSTRUMENTED DIALOG: ", "{that}.options.gradeNames"]
+        }
+    }
+});
+
+gpii.tests.app.instrumentedDialog.requestDialogCoverage = function (dialog) {
+    var sendCoverageCommand = fluid.stringTemplate("(%function)()", { function: gpii.tests.app.sendRendererCoverage.toString() });
+    gpii.test.executeJavaScript(dialog, sendCoverageCommand);
+};
+
+/**
+ * Request coverage data from all currently created BrowserWindows. Continues with
+ * tests execution once all windows have notified that their coverage had been sent successfully, 
+ *
+ * Note that the coverage is collected by executing a command sent from the main process - `gpii.tests.app.sendRendererCoverage`. 
+ * For more details refer to gpii.test.executeJavaScript.
+ */
+gpii.tests.app.instrumentedDialog.requestCoverage = function () {
```

Comment:
This really shouldn't be necessary unless you have test fixtures that use something other than QUnit.  You can have the client hook itself into the QUnit test lifecycle and send results on test suite completion.

Reply:
We don't run tests in the BrowserWindows themselves and therefore we don't have QUnit there. Instead we simulate interactions in our "main process" tests and that way we test most of the basic functionality of the BrowserWindows.